### PR TITLE
added depth criteria

### PR DESCRIPTION
**Pull request by dmitrykazanbaev**

### CODE REVIEW

File: pywinauto/uia_element_info.py [hunk 1]
```diff
@@ -259,15 +259,34 @@ def children(self, **kwargs):
...
         cond = IUIA().build_condition(**kwargs)
-        return self._get_elements(IUIA().tree_scope["descendants"], cond, cache_enable)
+        elements = self._get_elements(IUIA().tree_scope["descendants"], cond, cache_enable)
+
+        if depth is not None:
+                if isinstance(depth, int) and depth > 0:
+                    elements = [element for element in elements if element.has_depth(self, depth)]
+                else:
+                    raise Exception("Depth must be natural number")
+
```
Comment (airelil) [inline pywinauto/uia_element_info.py]: Same here, DRY. The whole "if" statement can be moved into a separate protected function

File: pywinauto/unittests/test_uia_element_info.py [hunk 2]
```diff
@@ -62,6 +62,22 @@ def testChildren(self):
             """Test whether a list of only immediate children of the element is equal"""
             self.assertEqual(len(self.ctrl.children()), 5)
 
+        def testDefaultDepthDescendants(self):
+            """Test whether a list of descendants with default depth of the element is equal"""
+            self.assertEqual(len(self.ctrl.descendants()), 69)
```
Comment (airelil) [inline pywinauto/unittests/test_uia_element_info.py]: I'm not big fan of such hard-coded numbers. The amount of controls can be changed later. Can we think  about something more dynamic ?

File: pywinauto/unittests/test_uia_element_info.py [hunk 3]
```diff
@@ -62,6 +62,22 @@ def testChildren(self):
...
+
+        def testDepthLevelOneDescendants(self):
+            """Test whether a list of descendants with depth=1 of the element is equal to children set"""
+            self.assertEqual(len(self.ctrl.descendants(depth=1)), len(self.ctrl.children()))
```
Comment (airelil) [inline pywinauto/unittests/test_uia_element_info.py]: That's good.

File: pywinauto/unittests/test_uia_element_info.py [hunk 4]
```diff
@@ -62,6 +62,22 @@ def testChildren(self):
...
+
+        def testDepthLevelThreeDescendants(self):
+            """Test whether a list of descendants with depth=3 of the element is equal"""
+            self.assertEqual(len(self.ctrl.descendants(depth=3)), 49)
```
Comment (airelil) [inline pywinauto/unittests/test_uia_element_info.py]: Again, a hard-coded number that also hard to verify. I'd, at least, looked up for a control with a small number of descendants.

File: pywinauto/unittests/test_uia_element_info.py (hunk 2 above)
Comment (dmitrykazanbaev) [inline pywinauto/unittests/test_uia_element_info.py]: I just wanted to keep the project code style :) See that previous test

**Comment by dmitrykazanbaev** [comment]
@airelil took into account your notes and dealt with them

**Comment by vasily-v-ryabov** [comment]
Hi @dmitrykazanbaev, first you need to fix unit tests for Python 2.7. There are 3 failures (while Py3.3+ are OK).
Just click on "Details" link against "AppVeyor build failed" at the bottom of the pull request page. Then dive into failed lines for Py2.7 tests.

File: pywinauto/element_info.py [hunk 5]
```diff
@@ -119,6 +119,26 @@ def children(self, **kwargs):
         """Return children of the element"""
         raise NotImplementedError()
 
+    def has_depth(self, root, depth):
```
Comment (vasily-v-ryabov) [inline pywinauto/element_info.py]: Doc string for every new method is a must.

File: pywinauto/unittests/test_uia_element_info.py [hunk 6]
```diff
@@ -62,6 +62,34 @@ def testChildren(self):
             """Test whether a list of only immediate children of the element is equal"""
             self.assertEqual(len(self.ctrl.children()), 5)
 
+        def testDefaultDepthDescendants(self):
```
Comment (vasily-v-ryabov) [inline pywinauto/unittests/test_uia_element_info.py]: Please use PEP-8 style for new test cases. Example: `test_default_depth_descendants`

**Comment by dmitrykazanbaev** [comment]
@vasily-v-ryabov fixed

**Pull request merged**